client/src/models/organizational_entities.js

```diff
@@ -0,0 +1,396 @@
+    this.name = name;
+    this.description = "";
+    this.orgs = [];
+  }
+};
+
+const Dept = class Dept extends static_subject_store_with_API_data(){
+  static lookup(org_id){
+    return super.lookup(
+      _.isNaN(+org_id) ?
+      org_id :
+      +org_id
+    );
+  }
+  static get type_name() { return 'dept'; }
+  static get singular(){ return trivial_text_maker("org"); }
+  static get plural(){ return trivial_text_maker("orgs");}
+  static depts_with_data(){ 
+    //lazy initialized
+    if(!this._depts_with_data){ 
+      this._depts_with_data = _.filter(
+        this.get_all(),
+        dept => !_.isEmpty(dept.table_ids)
+      );
+    }
+    return this._depts_with_data;
+  }
+  static depts_without_data(){ 
+    //lazy initialized
+    if(!this._depts_without_data){ 
+      this._depts_without_data = _.filter(
+        this.get_all(),
+        dept => _.isEmpty(dept.table_ids)
+      );
+    }
+    return this._depts_without_data;
+  }
+  static create_and_register(def){
+    const inst = new Dept(def);
+    this.register(inst.id,inst);
+    if(!_.isEmpty(inst.dept_code)){
+      this.register(inst.dept_code,inst);
+    }
+    return inst;
+  }
+  constructor(def){
+    super();
+    Object.assign(
+      this, 
+      {
+        name: def.legal_name,
+        id: def.unique_id,
+        minister_objs: [],
+        table_ids: [],
+        crsos: [],
+      },
+      def
+    );
+  }
+  get is_first_wave(){ // delete on drr17 exit
+    return this.dp_status === 'fw';
+  }
+
+  get programs(){
+    return _.chain(this.crsos)
+      .map('programs')
+      .flatten()
+      .compact()
+      .value();
+  }
+  // TODO: pipeline will change to set programs has_planned_spending to 0/1 (based on whether they have a stamped DP, and CRs and depts will roll this up (using _.some()))
+  // wait on has_planned_spending and dp_status cleanup until then
+  get has_planned_spending(){
+    return !(
+      _.includes([
+        "CSEC", 
+        "CSIS",
+        "FCAC",
+        "IJC",
+        "GG",
+      ],this.dept_code) ||
+      _.includes([ 
+        "Crown Corporations", 
+        "Sociétés d'État", 
+        "Parliamentary Entities",
+        "Entités Parlementaires",
+      ],this.inst_form)
+    );
+  }
+  get is_rpp_org(){
+    return this.dp_status !== false;
+  }
+  get dp_status(){ 
+    const val = this._dp_status;
+    if(val === 1){ 
+      return "fw";
+    } else if(val === 0){
+      return "sw";
+    } else {
+      return false;
+    }
+  }
+  get fancy_name(){
+    return this.applied_title || this.name;
+  }
+  get old_name(){
+    return this.old_applied_title;
+  }
+  get tables(){
+    return this.table_ids; 
+  }
+  get min(){
+    return this.ministry.name;
+  }
+  get minister(){
+    return _.map(this.ministers, 'name');
+  }
+  get auditor(){
+    return this.auditor_str;
+  }
+  get legislation(){
+    return (
+      _.isEmpty(this._legislation) ?
+      [] :
+      [ this._legislation ]
+    );
+  }
```

Comment:
`legislation` getter's not needed anymore, and `populate_stores.js` can now directly set the `legislation` property instead of renaming it `_legislation`.

Reply:
:+1: